rfcs/0034-system-parameters.md

```diff
@@ -0,0 +1,115 @@
+---
+feature: system-parameters
+authors:
+  - "Eric Fu"
+start_date: "2023/01/11"
+---
+
+# RFC: System Parameters on Meta
+
+## Motivation
+
+Currently, we have 2 kinds of configurations:
+
+- Command-line arguments
+- Configuration file `risingwave.toml`
+
+[Distinguish system config variables and cli options #5676](https://github.com/risingwavelabs/risingwave/issues/5676) clarifies the difference between CLI args and config file:
+
+> Intuitively, *Options* should contain basic user-facing options, e.g., addr & host and maybe some feature gates, while *Config* contains deep-dark internal system variables to fine-tune the system performance.
+
+However, this is not a complete solution because
+
+1. **`risingwave.toml` is a local file**, which makes it possible that multiple nodes could see different parameters and this is absolutely unacceptable. For example, if some nodes use `data_directory = 'hummock_001'` and others use `data_directory = 'hummock_002'`, the stored data will be corrupted.
+2. **Unable to `SET` a config after the process started**. Even if changes are written to disk, new nodes still miss these changes.
+
+This proposal will continue to improve the configuration mechanism but also repudiate the design of #5676.
+
+Previously I think this might not be an urgent issue until I realized that our Cloud platform couples tightly with more and more configurations. The later we commence, the heavier burden it will become. 
+
+## Design
+
+```mermaid
+graph LR
+    all[Does it apply to the RisingWave instance?]
+    all ==No==> process["Process Configs (Local)"]
+    all ==Yes==> system["System Configs (on Meta)"]
+    process --> cli_args(Command-line arguments)
+    process --> conf_file("Config file (.toml)")
+    system --> sql_cmd(SQL interface: SET / SHOW)
+    system --> cloud(Web Console)
+```
+
+### System Configs
```

Comment:
I think we need to clarify: Are system configs distinct from config files? From my understanding of the motivation it's yes. A related question is: If so, the default values for them (i.e., the values at startup) cannot be changed, right? 🤔

But since you referred PG, in PG they are the same set of configs:

> These variables can be set using the `SET` statement, by editing the `postgresql.conf` configuration file, through the `PGOPTIONS` environmental variable (when using libpq or a libpq-based application), or through command-line flags when starting the `postgres` server.

Reply:
Things are easier for single-node systems like PG/MySQL. Here we need to to either 1) move all configs to meta services  2) divided them into 2 sets: local and global. 

I chose 2 because 1 is impossible to avoid. For example, `--memory` must be a local parameter.